File: java/doit_java/Chapter6/src/cooperation/Subway.java
// 지하철 클래스 구현하기
package cooperation;

public class Subway {
    String lineNumber;  // 지하철 노선
    int passengerCount; // 승객수
    int money;          // 버스 수입

    // 지하철 노선 번호를 매개변수로 받는 생성자
    public Subway(String lineNumber) {
        this.lineNumber = lineNumber;
    }
    // 승객이 지하철에 탄 경우를 구현한 메서드
    public void take(int money) {
        this.money += money;    // 지하철 수입 증가
        passengerCount++;       // 승객수 증가
    }
    // 지하철 정보를 출력하는 메서드
    public void showInfo() {
        System.out.println(lineNumber + "의 승객은 " + passengerCount + "명이고, 수입은 " + money + "원 입니다.");
    }
}

File: python/python_oop_practice/p_oop_2/chap05/cooperation/class_module/subway.py
class Subway:
    def __init__(self, line_number):
        self.line_number = line_number
        self.money = 0
        self.passenger_count = 0

    def take(self, money):
        self.money += money
        self.passenger_count += 1

    def show_subway_info(self):
        print(str(self.line_number) + "호선 지하철의 승객은 " + str(self.passenger_count) \
        + "명 이고, 수입은 " + str(self.money) + "원 입니다.")
